rfd/0139-gogo-must-go.md

```diff
@@ -116,12 +151,128 @@ casts or conversions within go code. We will have to take on this conversion wor
 ##### `gogoproto.nullable`
 
 We will need to remove the `gogoproto.nullable` tags, which overall shouldn't be a huge impact.
-We should migrate to using `optional` as defined [here](https://protobuf.dev/programming-guides/proto3/#field-labels).
+We will need to rely on our `CheckAndSetDefaults` functions, which should generally handle this
+today.
 
 ##### `gogoproto.stdtime`
 
 We will need to migrate to using `timestamppb` directly instead of relying on gogo's `stdtime` tag.
 
+### Proposed solution
+
+#### Create internal objects
+
+Currently, in `api/types` we define a large number of interfaces that are subsequently used to wrap
```

Comment:
Counter proposal: could we take this chance to refactor those interfaces out of existence? They are one of the [less idiomatic](https://github.com/golang/go/wiki/CodeReviewComments#interfaces) parts of Teleport and serve little purpose.